client/src/connection/rest/server.ts

```diff
@@ -0,0 +1,133 @@
+import { ServersApi, Server, Link } from "./api/compute";
+import { ComputeSession } from "./session";
+import { AxiosResponse } from "axios";
+
+export class ComputeServer extends Compute {
+  api;
+  _self: Server & BaseCompute;
+
+  constructor(id: string) {
+    super();
+
+    this._self = { id: id };
+
+    this.api = ServersApi(getApiConfig());
+  }
+
+  get id(): string {
+    return this._self?.id || "";
+  }
+
+  get links(): Array<Link> {
+    return this._self?.links || [];
+  }
+
+  static fromInterface(server: Server): ComputeServer {
+    const _server = new ComputeServer("");
+    _server._self = server;
+    return _server;
+  }
+
+  static fromResponse(response: AxiosResponse): ComputeServer {
+    const _server = ComputeServer.fromInterface(response.data);
+    _server.etag = response.headers.etag;
+    return _server;
+  }
+
+  async self<Server>(): Promise<Server> {
+    if (this._self.id === undefined) {
+      throw new Error("Cannot call self on object with no id");
+    }
+
+    const res = await this.api.getServer({ serverId: this.id });
+    if (res.status === 200) {
+      this._self = res.data;
+      this.etag = res.headers.etag;
+      return res.data;
+    } else {
+      throw new Error(
+        `Error getting server with ID  ${this.id} - ${res.message}`
+      );
+    }
+  }
+
+  async createSession(): Promise<ComputeSession> {
+    if (this._self.links === undefined) {
+      await this.self();
+    }
+
+    const link = this.getLink(this.links, "createSession");
+    if (link === undefined) {
+      throw new Error("Server does not have createSession link");
+    }
+
+    //Create the session
+    //TODO: Add session create options
+    //TODO: Session request should be an interface
+    const body = {
+      version: 1,
+      name: "mysess",
+      description: "This is a session",
+      attributes: {},
+      environment: {
+        options: ["-validmemname EXTEND", "-validvarname ANY", "-memsize 0"],
+        autoExecLines: [],
+      },
+    };
+
+    const resp = await this.requestLink(link, { data: body });
+
+    if (!(resp.status === 201 || resp.status === 200)) {
+      throw new Error(<string>resp?.data);
+    }
+
+    //Create the session from the http resposne
+    const session = ComputeSession.fromResponse(resp);
+    return session;
+  }
+
+  async getSession(sessionId?: string): Promise<ComputeSession> {
+    if (sessionId !== undefined) {
+      return new ComputeSession(sessionId);
+    } else {
+      return this.createSession();
+    }
+  }
+
+  async getState(options?: stateOptions): Promise<string> {
+    if (this._self.links === undefined) {
+      await this.self();
+    }
+
+    const params = {};
+    const headers = {};
+
+    if (options !== undefined) {
+      if (options.onChange) {
+        headers["If-None-Match"] = this.etag;
+      }
+
+      if (options.wait) {
+        // eslint-disable-next-line @typescript-eslint/dot-notation
```

Comment:
I think this is probably fine to be `params.timeout`

Reply:
From the previous comment I made about dot notation, it looks odd to use dot notation in this if statement, but type notation in the one above it.

Reply:
I'd advocate for following the eslint rule here instead of adding exceptions. It's somewhat inconsistent in look here, but it's consistent with the eslint rule and similar to your change in common.ts